src/redux/features/deliverables/deliverablesSlice.ts

```diff
@@ -30,30 +30,23 @@ export const deliverablesSearchReducer = deliverablesListSlice.reducer;
  * Can be accessed by a request ID or by a deliverable/project ID string
  */
 const initialStateDeliverables: { [key: number | string]: StatusT<Deliverable> } = {};
-export const makeDeliverableProjectIdKey = (deliverableId: number, projectId: number) =>
-  `d${deliverableId}-p${projectId}`;
+
+type DeliverableProjectIdArg = { deliverableId: number; projectId: number };
+export const deliverableCompositeKeyFn = (arg: unknown): string => {
+  const castArg = arg as DeliverableProjectIdArg;
```

Comment:
I went this route as opposed to using `any` in the `buildReducers` function or forcing another generic type. We explicitly check for empty string in the `setStatus` function before using the composite key to store the result in state.